test/functional/test_framework/test_framework.py

```diff
@@ -444,6 +445,9 @@ def set_test_params(self):
         self.num_nodes = 2
         self.setup_clean_chain = True
 
+    def run_test(self):
```

Comment:
To more strictly enforce this protocol I had to add this here, even though it just calls the superclass implementation anyway. I figured this is cleaner than making an exception in the metaclass

Reply:
That's fine. ComparisonTestFramework should hopefully go away very soon (#11818)

Reply:
Needs rebase